### PR TITLE
Core/Spells: Fix periodic rolling adding bonuses twice

### PR DESCRIPTION
**Pull request by HKevinH**
### Changes proposed

-  Cherry Picked TCPP
-  Core/Spells: Fix periodic rolling adding bonuses twice
Calculation is now done in CalculateAmount

### Issues addressed

- Closes

### SOURCE:

- https://github.com/The-Cataclysm-Preservation-Project/TrinityCore/commit/1fe5563431fc722154f5138c71c69c177cfb3470#diff-a2a2343f2b1620e61594cb8b19343ebddb437f761cb9ff6a6ea26f36b038efa5

- Thank you for collaborating with the project.
- For an efficient working methodology.
- It is obvious to make long pull request, unless it is necessary.
- A short pull request would be easy to test and approve.

### CODE REVIEW

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 1]
```diff
@@ -450,10 +451,7 @@ AuraEffect::AuraEffect(Aura* base, uint8 effIndex, int32 const* baseAmount, Unit
     CalculateSpellMod();
 }
 
-AuraEffect::~AuraEffect()
-{
-    delete m_spellmod;
-}
+AuraEffect::~AuraEffect() { delete m_spellmod; }
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 2]
```diff
@@ -1477,7 +1475,7 @@ void AuraEffect::HandleModInvisibilityDetect(AuraApplication const* aurApp, uint
 
     // call functions which may have additional effects after chainging state of unit
     if (target->IsInWorld())
-    target->UpdateObjectVisibility();
+        target->UpdateObjectVisibility();
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:1481]: Code-styling, brackets.

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 3]
```diff
@@ -1541,7 +1539,7 @@ void AuraEffect::HandleModInvisibility(AuraApplication const* aurApp, uint8 mode
         target->RemoveAurasWithInterruptFlags(SpellAuraInterruptFlags::StealthOrInvis);
     }
     if (target->IsInWorld())
-    target->UpdateObjectVisibility();
+        target->UpdateObjectVisibility();
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:1545]: Code-styling, brackets.

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 4]
```diff
@@ -1567,7 +1565,7 @@ void AuraEffect::HandleModStealthDetect(AuraApplication const* aurApp, uint8 mod
 
     // call functions which may have additional effects after chainging state of unit
     if (target->IsInWorld())
-    target->UpdateObjectVisibility();
+        target->UpdateObjectVisibility();
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:1571]: Code-styling, brackets

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 5]
```diff
@@ -1609,7 +1607,7 @@ void AuraEffect::HandleModStealth(AuraApplication const* aurApp, uint8 mode, boo
     }
 
     if (target->IsInWorld())
-    target->UpdateObjectVisibility();
+        target->UpdateObjectVisibility();
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:1613]: Code-styling, brackets

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 6]
```diff
@@ -1627,7 +1625,7 @@ void AuraEffect::HandleModStealthLevel(AuraApplication const* aurApp, uint8 mode
 
     // call functions which may have additional effects after chainging state of unit
     if (target->IsInWorld())
-    target->UpdateObjectVisibility();
+        target->UpdateObjectVisibility();
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:1631]: Code-styling, brackets

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 7]
```diff
@@ -2639,7 +2637,7 @@ void AuraEffect::HandleAuraModStalked(AuraApplication const* aurApp, uint8 mode,
...
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:2643]: Code-styling, brackets

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 8]
```diff
@@ -5243,7 +5232,7 @@ void AuraEffect::HandleAuraModFakeInebriation(AuraApplication const* aurApp, uin
...
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:5238]: Code-styling, brackets

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 9]
```diff
@@ -5986,7 +5975,7 @@ void AuraEffect::HandlePeriodicHealthLeechAuraTick(Unit* target, Unit* caster) c
     // SendSpellNonMeleeDamageLog expects non-absorbed/non-resisted damage
     if (caster)
 
-    caster->SendSpellNonMeleeDamageLog(target, GetId(), damage, GetSpellInfo()->GetSchoolMask(), absorb, resist, false, 0, crit);
+        caster->SendSpellNonMeleeDamageLog(target, GetId(), damage, GetSpellInfo()->GetSchoolMask(), absorb, resist, false, 0, crit);
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:5981]: Code-styling, brackets and whitespace on line 5977.

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 10]
```diff
@@ -3961,10 +3953,7 @@ void AuraEffect::HandleAuraModBlockPercent(AuraApplication const* aurApp, uint8
     target->ToPlayer()->UpdateBlockPercentage();
 }
 
-void AuraEffect::HandleAuraModRegenInterrupt(AuraApplication const* aurApp, uint8 mode, bool apply) const
-{
-    HandleModManaRegen(aurApp, mode, apply);
-}
+void AuraEffect::HandleAuraModRegenInterrupt(AuraApplication const* aurApp, uint8 mode, bool apply) const { HandleModManaRegen(aurApp, mode, apply); }
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:3959]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 11]
```diff
@@ -3749,10 +3744,7 @@ void AuraEffect::HandleModPowerRegen(AuraApplication const* aurApp, uint8 mode,
     aurApp->GetTarget()->UpdatePowerRegeneration(Powers(GetMiscValue()));
 }
 
-void AuraEffect::HandleModPowerRegenPCT(AuraApplication const* aurApp, uint8 mode, bool apply) const
-{
-    HandleModPowerRegen(aurApp, mode, apply);
-}
+void AuraEffect::HandleModPowerRegenPCT(AuraApplication const* aurApp, uint8 mode, bool apply) const { HandleModPowerRegen(aurApp, mode, apply); }
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:3750]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 12]
```diff
@@ -3138,10 +3136,7 @@ void AuraEffect::HandleAuraModIncreaseSpeed(AuraApplication const* aurApp, uint8
     target->UpdateSpeed(MOVE_RUN);
 }
 
-void AuraEffect::HandleAuraModIncreaseMountedSpeed(AuraApplication const* aurApp, uint8 mode, bool apply) const
-{
-    HandleAuraModIncreaseSpeed(aurApp, mode, apply);
-}
+void AuraEffect::HandleAuraModIncreaseMountedSpeed(AuraApplication const* aurApp, uint8 mode, bool apply) const { HandleAuraModIncreaseSpeed(aurApp, mode, apply); }
```
Comment (Meltie2013) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:3142]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Spell.cpp [hunk 13]
```diff
@@ -341,15 +337,9 @@ Corpse* SpellCastTargets::GetCorpseTarget() const
     return nullptr;
 }
 
-WorldObject* SpellCastTargets::GetObjectTarget() const
-{
-    return m_objectTarget;
-}
+WorldObject* SpellCastTargets::GetObjectTarget() const { return m_objectTarget; }
```
Comment (Meltie2013) [inline src/server/game/Spells/Spell.cpp:340]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Spell.cpp [hunk 14]
```diff
@@ -341,15 +337,9 @@ Corpse* SpellCastTargets::GetCorpseTarget() const
...
 
-ObjectGuid SpellCastTargets::GetObjectTargetGUID() const
-{
-    return m_objectTargetGUID;
-}
+ObjectGuid SpellCastTargets::GetObjectTargetGUID() const { return m_objectTargetGUID; }
```
Comment (Meltie2013) [inline src/server/game/Spells/Spell.cpp:342]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Spell.cpp [hunk 15]
```diff
@@ -387,15 +377,9 @@ void SpellCastTargets::UpdateTradeSlotItem()
     }
 }
 
-SpellDestination const* SpellCastTargets::GetSrc() const
-{
-    return &m_src;
-}
+SpellDestination const* SpellCastTargets::GetSrc() const { return &m_src; }
```
Comment (Meltie2013) [inline src/server/game/Spells/Spell.cpp:380]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Spell.cpp [hunk 16]
```diff
@@ -387,15 +377,9 @@ void SpellCastTargets::UpdateTradeSlotItem()
...
 
-Position const* SpellCastTargets::GetSrcPos() const
-{
-    return &m_src._position;
-}
+Position const* SpellCastTargets::GetSrcPos() const { return &m_src._position; }
```
Comment (Meltie2013) [inline src/server/game/Spells/Spell.cpp:382]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Spell.cpp [hunk 17]
```diff
@@ -421,20 +405,11 @@ void SpellCastTargets::ModSrc(Position const& pos)
     m_src.Relocate(pos);
 }
 
-void SpellCastTargets::RemoveSrc()
-{
-    m_targetMask &= ~(TARGET_FLAG_SOURCE_LOCATION);
-}
+void SpellCastTargets::RemoveSrc() { m_targetMask &= ~(TARGET_FLAG_SOURCE_LOCATION); }
```
Comment (Meltie2013) [inline src/server/game/Spells/Spell.cpp]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Spell.cpp [hunk 18]
```diff
@@ -421,20 +405,11 @@ void SpellCastTargets::ModSrc(Position const& pos)
...
 
-SpellDestination const* SpellCastTargets::GetDst() const
-{
-    return &m_dst;
-}
+SpellDestination const* SpellCastTargets::GetDst() const { return &m_dst; }
```
Comment (Meltie2013) [inline src/server/game/Spells/Spell.cpp]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Spell.cpp [hunk 19]
```diff
@@ -421,20 +405,11 @@ void SpellCastTargets::ModSrc(Position const& pos)
...
 
-WorldLocation const* SpellCastTargets::GetDstPos() const
-{
-    return &m_dst._position;
-}
+WorldLocation const* SpellCastTargets::GetDstPos() const { return &m_dst._position; }
```
Comment (Meltie2013) [inline src/server/game/Spells/Spell.cpp]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Spell.cpp [hunk 20]
```diff
@@ -478,20 +453,11 @@ void SpellCastTargets::ModDst(SpellDestination const& spellDest)
     m_dst = spellDest;
 }
 
-void SpellCastTargets::RemoveDst()
-{
-    m_targetMask &= ~(TARGET_FLAG_DEST_LOCATION);
-}
+void SpellCastTargets::RemoveDst() { m_targetMask &= ~(TARGET_FLAG_DEST_LOCATION); }
```
Comment (Meltie2013) [inline src/server/game/Spells/Spell.cpp:464]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Spell.cpp [hunk 21]
```diff
@@ -478,20 +453,11 @@ void SpellCastTargets::ModDst(SpellDestination const& spellDest)
...
 
-bool SpellCastTargets::HasSrc() const
-{
-    return (GetTargetMask() & TARGET_FLAG_SOURCE_LOCATION) != 0;
-}
+bool SpellCastTargets::HasSrc() const { return (GetTargetMask() & TARGET_FLAG_SOURCE_LOCATION) != 0; }
```
Comment (Meltie2013) [inline src/server/game/Spells/Spell.cpp:466]: Code-styling

Should keep the same styling as before.

File: src/server/game/Spells/Spell.cpp [hunk 22]
```diff
@@ -478,20 +453,11 @@ void SpellCastTargets::ModDst(SpellDestination const& spellDest)
...
 
-bool SpellCastTargets::HasDst() const
-{
-    return (GetTargetMask() & TARGET_FLAG_DEST_LOCATION) != 0;
-}
+bool SpellCastTargets::HasDst() const { return (GetTargetMask() & TARGET_FLAG_DEST_LOCATION) != 0; }
```
Comment (Meltie2013) [inline src/server/game/Spells/Spell.cpp:468]: Code-styling

Should keep the same styling as before.

**Review by Meltie2013 — CHANGES_REQUESTED** [review]
There are some changes I like to see here, but there are over ~45 points that should be reviewed. Most code with brackets being shifted into one line, should remain the same as before since there were no changes specifically.

@seobryn @acidmanifesto 
Would like you input on this.

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp (hunk 1 above)
Comment (HKevinH) [reply src/server/game/Spells/Auras/SpellAuraEffects.cpp]: Sorry, My editor has the prettier so that when I save it, I leave it in place.

Comment (Meltie2013) [reply src/server/game/Spells/Auras/SpellAuraEffects.cpp]: No worries

**Review by acidmanifesto — APPROVED** [review]
Tested working by asmaduex. Appropriately referenced. Workflow pass. I am being requested to merge this and will do so.

**Pull request merged**